packages/react-native-updater/android/src/main/java/app/capgo/rnupdater/CapgoUpdaterModule.kt

```
@@ -0,0 +1,275 @@
+        ) { percent ->
+          val event = Arguments.createMap()
+          event.putInt("percent", percent)
+          event.putMap("bundle", bundleToMap(pending.copy(status = "downloading")))
+          emit("download", event)
+        }
+
+        val complete = Arguments.createMap()
+        complete.putMap("bundle", bundleToMap(record))
+        emit("downloadComplete", complete)
+        promise.resolve(bundleToMap(record))
+      } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)